File: python/lc3597.py
class Solution:
    def partitionString(self, s: str) -> List[str]:
        seen = set()
        output = []
        cur = ""
        for c in s:
            cur += c
            if cur not in seen:
                seen.add(cur)
                output.append(cur)
                cur = ""
        return output

File: cs/lc3597.cs
public class Solution {
    public IList<string> PartitionString(string s) {
        var seen = new HashSet<string>();
        var output = new List<string>();
        String cur = "";
        foreach(var c in s)
        {
            cur += c;            
            if (!seen.Contains(cur))
            {
                seen.Add(cur);
                output.Add(cur);
                cur = "";
            }
        }
        return output;
    }
}
